bad class file for SPacketPlayerListItem$AddPlayerData due to bad RuntimeInvisibleParameterAnnotations attribute, when setupDecompWorkspace has not been run

This has already been reported as MinecraftForge/MinecraftForge#3238, but it's still an issue, so I'm creating a new issue here to keep track of it.

> ```
> C:\Pokechu22\Git_\SetupCITest\build\sources\main\java\com\mumfrey\DummyFile.java:31: error: cannot access AddPlayerData
>     public final SPacketPlayerListItem.AddPlayerData getEntry() {
>                                       ^
>   bad class file: C:\Users\Pokechu22\.gradle\caches\minecraft\net\minecraft\minecraft\1.12.2\snapshot\20160517\minecraftBin-1.12.2.jar(net/minecraft/network/play/server/SPacketPlayerListItem$AddPlayerData.class)
>     bad RuntimeInvisibleParameterAnnotations attribute: AddPlayerData(GameProfile,int,GameType,ITextComponent)
>     Please remove or make sure it appears in the correct subdirectory of the classpath.
> ```

To reproduce, use https://github.com/Mumfrey/SetupCITest/ (edit the versions in it, though, to be 1.12.2 and FG 2.3), and run `gradle build` without `setupDecompWorkspace`.  The above error should occur.

OK, here's a diff of the constructor to `SPacketPlayerListItem`, as given through `javap -v`.  `-` is `bin`, `+` is `src`.

```diff
   public net.minecraft.network.play.server.SPacketPlayerListItem$AddPlayerData(net.minecraft.network.play.server.SPacketPlayerListItem, com.mojang.authlib.GameProfile, int, net.minecraft.world.GameType, net.minecraft.util.text.ITextComponent);
     descriptor: (Lnet/minecraft/network/play/server/SPacketPlayerListItem;Lcom/mojang/authlib/GameProfile;ILnet/minecraft/world/GameType;Lnet/minecraft/util/text/ITextComponent;)V
     flags: ACC_PUBLIC
     Code:
       stack=2, locals=6, args_size=6
          0: aload_0
          1: aload_1
-         2: putfield      #33                 // Field field_179968_a:Lnet/minecraft/network/play/server/SPacketPlayerListItem;
+         2: putfield      #1                  // Field this$0:Lnet/minecraft/network/play/server/SPacketPlayerListItem;
          5: aload_0
-         6: invokespecial #36                 // Method java/lang/Object."<init>":()V
+         6: invokespecial #2                  // Method java/lang/Object."<init>":()V
          9: aload_0
         10: aload_2
-        11: putfield      #38                 // Field profile:Lcom/mojang/authlib/GameProfile;
+        11: putfield      #3                  // Field profile:Lcom/mojang/authlib/GameProfile;
         14: aload_0
         15: iload_3
-        16: putfield      #40                 // Field ping:I
+        16: putfield      #4                  // Field ping:I
         19: aload_0
         20: aload         4
-        22: putfield      #42                 // Field gamemode:Lnet/minecraft/world/GameType;
+        22: putfield      #5                  // Field gamemode:Lnet/minecraft/world/GameType;
         25: aload_0
         26: aload         5
-        28: putfield      #44                 // Field displayName:Lnet/minecraft/util/text/ITextComponent;
+        28: putfield      #6                  // Field displayName:Lnet/minecraft/util/text/ITextComponent;
         31: return
+      LineNumberTable:
+        line 197: 0
+        line 198: 9
+        line 199: 14
+        line 200: 19
+        line 201: 25
+        line 202: 31
       LocalVariableTable:
         Start  Length  Slot  Name   Signature
             0      32     0  this   Lnet/minecraft/network/play/server/SPacketPlayerListItem$AddPlayerData;
-            0      32     1 p_i46663_1_   Lnet/minecraft/network/play/server/SPacketPlayerListItem;
-            0      32     2 p_i46663_2_   Lcom/mojang/authlib/GameProfile;
-            0      32     3 p_i46663_3_   I
-            0      32     4 p_i46663_4_   Lnet/minecraft/world/GameType;
-            0      32     5 p_i46663_5_   Lnet/minecraft/util/text/ITextComponent;
-      LineNumberTable:
-        line 187: 0
-        line 188: 9
-        line 189: 14
-        line 190: 19
-        line 191: 25
-        line 192: 31
+            0      32     1 this$0   Lnet/minecraft/network/play/server/SPacketPlayerListItem;
+            0      32     2 profileIn   Lcom/mojang/authlib/GameProfile;
+            0      32     3 latencyIn   I
+            0      32     4 gameModeIn   Lnet/minecraft/world/GameType;
+            0      32     5 displayNameIn   Lnet/minecraft/util/text/ITextComponent;
     RuntimeVisibleParameterAnnotations:
       parameter 0:
       parameter 1:
       parameter 2:
       parameter 3:
-        0: #31()
-      parameter 4:
-        0: #31()
+        0: #42()
```

Hm, yeah, that does look incorrect; `bin` has 5 parameters while `src` has 4.  I do see 5 parameters on the method, but one of them is the synthetic `this$0` parameter for the enclosing class.  [The JVMS](https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.7.18 "§4.7.18 The RuntimeVisibleParameterAnnotations Attribute") says that "The value of the num_parameters item gives the number of formal parameters of the method represented by the method_info structure on which the annotation occurs." (and that "This duplicates information that could be extracted from the method descriptor.", which seems to be what the check I found before is doing).  I'm not fully sure whether that synthetic thing counts as a formal parameter.  I'll investigate that further.

Let's check the unobf'd version of the class, `jp$b`.  ... it has it too.

```java
class Why {
	public jp.b getAddPlayerData() {
		return null;
	}
}
```

Compile the above code with `javac -cp 1.12.2.jar Why.java` and you will get

> ```
> Why.java:2: error: cannot access b
>         public jp.b getAddPlayerData() {
>                  ^
>   bad class file: 1.12.2.jar(jp$b.class)
>     bad RuntimeInvisibleParameterAnnotations attribute: b(GameProfile,int,ams,hh)
>     Please remove or make sure it appears in the correct subdirectory of the classpath.
> ```

So, this isn't FG's fault.  But I don't know where else to go from here.  Only thing I know to do is to look into the formal parameters bit.

--- a/build.gradle
+++ b/build.gradle
@@ -84,7 +84,7 @@ dependencies {
 
     // mcp stuff
     shade 'de.oceanlabs.mcp:RetroGuard:3.6.6'
-    shade('de.oceanlabs.mcp:mcinjector:3.4-SNAPSHOT'){
+    shade('de.oceanlabs.mcp:mcinjector:3.6'){
      exclude group: 'org.ow2.asm', module: 'asm-debug-all'
     }
     shade('net.minecraftforge.srg2source:Srg2Source:4.0-SNAPSHOT'){

